### PR TITLE
resampler: handle explicitly the case that in farmes less than out frames

### PR DESCRIPTION
**Pull request by achronop**

### CODE REVIEW

**Comment by achronop** [comment]
Hmm are we running the tests on travis? I've just tried them locally and I am getting a failure.

**Comment by achronop** [comment]
Yeah, no testing in Windows, that can be deceiving ...

File: src/cubeb_resampler.cpp [hunk 1]
```diff
@@ -80,6 +80,11 @@ long passthrough_resampler<T>::fill(void * input_buffer, long * input_frames_cou
       // to the callback
       internal_input_buffer.push(static_cast<T*>(input_buffer),
                                  frames_to_samples(*input_frames_count));
+      if (internal_input_buffer.length() < frames_to_samples(output_frames)) {
+        // Grrr glitch!!!
```
Comment (kinetiknz) [inline src/cubeb_resampler.cpp]: Better comment please :)

**Comment by achronop** [comment]
try run: https://treeherder.mozilla.org/#/jobs?repo=try&revision=a7bed10d6a388605c4b47152aea21b4a6b1113fc

File: src/cubeb_resampler.cpp (hunk 1 above)
Comment (padenot) [reply src/cubeb_resampler.cpp]: Yeah, same as below or something.

**Comment by achronop** [comment]
@nyorain, would you like to have a look at it. It's around your code so it would be nice to tell me what you think. Thanks.

File: src/cubeb_resampler.cpp [hunk 2]
```diff
@@ -91,6 +97,17 @@ long passthrough_resampler<T>::fill(void * input_buffer, long * input_frames_cou
       unsigned long samples_off = frames_to_samples(output_frames);
       internal_input_buffer.push(static_cast<T*>(input_buffer) + samples_off,
                                  frames_to_samples(*input_frames_count - output_frames));
+    } else if (*input_frames_count < output_frames) {
```
Comment (nyorain) [inline src/cubeb_resampler.cpp]: Not important, but it should be possible to combine this and the first `if` branch to remove some redundant code like this:
```c
if (internal_input_buffer.length() != 0 || *input_frames_count < output_frames)
```
The glitch case will already be correctly handled by the check you inserted into that branch I guess.

**Comment by nyorain** [comment]
Otherwise it looks fine. But the conditions you handle will trigger the `assert` at the beginning of the function, you probably want to change that.

File: src/cubeb_resampler.cpp (hunk 2 above)
Comment (achronop) [reply src/cubeb_resampler.cpp]: Yeah, I can do that, it will look cleaner.

**Comment by achronop** [comment]
Yeah, we have an extra patch in Firefox to disable that assert. I will merge it here instead of adding it on the top of an libcubeb import. Thank you for the comments.

**Pull request merged**